test-tests/shared/src/test/scala/zio/test/TestSpec.scala

```diff
@@ -60,6 +60,38 @@ object TestSpec extends ZIOBaseSpec {
         message <- STM.succeed("Hello from an STM transaction!")
       } yield assert(message)(anything)
     },
+    test("fail-fast assertion by automatic lifting to ZIO") {
+      for {
+        ref <- Ref.make(0)
+        spec = test("test") {
+                 for {
+                   _ <- ref.update(_ + 1)
+                   _ <- assertTrue(false)
+                   _ <- ref.update(_ + 1)
+                 } yield assertCompletes
+               }
+        summary <- execute(spec)
+        count   <- ref.get
+      } yield assert(count)(equalTo(1)) &&
+        assert(summary.fail)(equalTo(1)) &&
+        assert(summary.failureDetails)(containsString("Result was false"))
+    },
+    test("composed assertions are automatically lifted to ZIO") {
+      for {
+        ref <- Ref.make(0)
+        spec = test("test") {
+                 for {
+                   _ <- ref.update(_ + 1)
+                   _ <- assertTrue(true) && assertTrue(false)
```

Comment:
I'm more wondering if we do `assertTrue(somethingFalse) && assertTrue(someThingElseFalse)` do we get both failures in the error message? I think this is the case where we don't want to fail fast.

Reply:
Good point. Revised the test case.